server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4852,9 +4857,39 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
                 _tmplService.attachIso(tmpl.getId(), vm.getId());
             }
         }
+
+        // Add extraConfig to user_vm_details table
+        Account caller = CallContext.current().getCallingAccount();
+        Long callerId = caller.getId();
+        if (EnableAdditionalVmConfig.valueIn(callerId)) {
+            String extraConfig = cmd.getExtraConfig();
+            if (!Strings.isNullOrEmpty(extraConfig)) {
+                String decodedUrl;
+                try {
+                    decodedUrl = URLDecoder.decode(extraConfig, "UTF-8");
+                } catch (UnsupportedEncodingException e) {
+                    throw new CloudRuntimeException("Failed to decode URL string: " + e.getMessage());
+                }
+
+                if (vm.getType() == VirtualMachine.Type.User) {
+                    validateConfigOption(decodedUrl, callerId);
+                    _uservmDetailsDao.addDetail(vm.getId(), "extraConfig", decodedUrl, true);
```

Comment:
As mentioned before, the "extraConfig" could be replaced by a static variable

Reply:
Done